### PR TITLE
Improve trust score docs

### LINKED ISSUE #482: Trust score description is not clear enough
As a user that is not familiar with trust score concepts, I can't understand what this check does. 

image

### LINKED ISSUE #601: [DOCS] Trust score comparison example improvement
Currently, the trust score comparison notebook lacks some explanations.
Please follow the guidelines from this issue: #543 to improve it

### PR DESCRIPTION
**Pull request by noamzbr**
#### Reference Issues/PRs

closes #601 and closes #482 

#### What does this implement/fix? Explain your changes.
Adds the breast cancer dataset and refurbished TrustScoreComparison docstring and notebook.

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

File: deepchecks/checks/distribution/trust_score_comparison.py [hunk 1]
```diff
@@ -27,13 +27,14 @@
...
     The process is as follows:
-    * Pre-process the train and test data into scaled numerics.
-    * Train a TrustScore (https://arxiv.org/abs/1805.11783) regressor based on train data + label.
-    * Predict on test data using the model.
-    * Use TrustScore to score the prediction of the model.
+
+    #. Pre-process the train and test data into scaled numerics.
+    #. Train a TrustScore (https://arxiv.org/abs/1805.11783) regressor based on train data + label.
+    #. Predict on test data using the model.
+    #. Use TrustScore to score the prediction of the model.
```
Comment (ItayGabbay) [inline deepchecks/checks/distribution/trust_score_comparison.py:38]: How does it look in the API refernce? Is #. a valid syntax in RST?

File: deepchecks/checks/distribution/trust_score_comparison.py [hunk 2]
```diff
@@ -142,11 +143,14 @@ def transform_numpy_label(np_label: np.ndarray) -> np.ndarray:
         top_k = test_data_sample.head(self.n_to_show)
         bottom_k = test_data_sample.tail(self.n_to_show)
 
-        headnote = """<span>
-        Trust score measures the agreement between the classifier and a modified nearest-neighbor
-        classifier on the testing example. Higher values represent samples that are "close" to training examples with
+        headnote = r"""<span>
+        Trust score (https://arxiv.org/abs/1805.11783) roughly measures the following quantity:<br><br>
```
Comment (ItayGabbay) [inline deepchecks/checks/distribution/trust_score_comparison.py:147]: Worth referring to this issue #637

File: docs/source/examples/checks/distribution/trust_score_comparison.ipynb [hunk 3]
```diff
@@ -2,106 +2,135 @@
  "cells": [
```
Comment (ItayGabbay) [inline docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: <p>Something is wrong in the first sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/644/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (ItayGabbay) [inline docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: <p>I think this title doesn't follow the standards we defined</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/644/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (ItayGabbay) [inline docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: <p>Really good notebook!!!!!!!!!! we should promote it to be in the front page of the docs.</p><p> </p><p>just 1 thought: I wonder whether a multi-class usecase will be interesting here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/644/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (matanper) [reply docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: it's an okay notebook, I've seen better...

Comment (noamzbr) [reply docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: nothing beats the single feature contribution notebook.

About multiclass - it will look nice but will be a bit harder to show specific phenomena in the distribution graphs

File: deepchecks/checks/distribution/trust_score_comparison.py (hunk 2 above)
Comment (noamzbr) [reply deepchecks/checks/distribution/trust_score_comparison.py]: Right, updated. 
Closes #637

File: docs/source/examples/checks/distribution/trust_score_comparison.ipynb (hunk 3 above)
Comment (matanper) [inline docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: <p>a. Fix structure to fit the headers.</p><p>b. maybe we should mention when a check does not have condition?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/644/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (shir22) [reply docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: <p>:)</p>

File: docs/source/examples/checks/distribution/trust_score_comparison.ipynb [hunk 4]
```diff
@@ -2,106 +2,136 @@
...
```
Comment (shir22) [inline docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: <p>verity -&gt; variety</p><p> </p><ol><li>"Detecting "problematic samples" before labels become available - prediction with low probability are assumed to be more likely to fail.</li></ol><p>would change the end "with low probability are more likely to be wrong"</p><p> </p><p>"This is because the methods and loss functions used by these models are often not designed to produce actual probabilities, and model performance is measured by metric that mostly rely only on predicted class. " -&gt; last part unclear</p><p> </p><p>--</p><p>link to paper - is it external?</p><p>--</p><p>last point (2):</p><ol><li>Identifying a degradation between the trust score on the test data when comparing it to the training data, which may indicate that the model will perform worse on test compared to train and serves as a method to detect concept drift. This condition is useful especially for cases when the test labels are not available, such as when performing inference on new and unknown data.</li></ol><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/644/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (shir22) [inline docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: <p>alternative phrasing (fixing some typos etc.) for last paragraph:</p><p>Furthermore, we may notice some other common characteristics, such as the fact that <code>worst texture</code> and <code>mean texture</code> both seem to be lower in the top scoring samples, while the worst scoring samples have <code>worst texture</code> and <code>mean texture</code>  values, both features with high feature importance for the AdaBoost model. Might it be that high texture samples are getting worse predictions by the model? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/644/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (shir22) [inline docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: <p>First, we introduce concept drift into the data by changing the relation between the <code>worst texture</code>  and <code>mean concave points</code>   features, both important features for the model. <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/644/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (noamzbr) [reply docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: <p>yes links are external </p>

Comment (shir22) [inline docs/source/examples/checks/distribution/trust_score_comparison.ipynb:2]: <ol><li>Detecting "problematic samples" before labels become available - prediction with low probability are more likely to be wrong.</li></ol><p> </p><p>prediction -&gt; predictions</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/644/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Review by shir22 — APPROVED** [review]
fix before merge one small typo
